### ISSUE TITLE
Coordinate/node size resolution for non-default axes regressed since 0.4.1

### ISSUE THREAD

**Issue by SillyFreak** [Jollywatt/typst-fletcher#66]
Setting `diagram(axes: )` allows one to interpret coordinates of the form `a, b` differently, e.g. to quickly switch betwen a left-to-right to top-to-bottom layout, or to flip axes. It seems that since 0.4.1, some code that does not respect these axes has made it into Fletcher.

<details>
<summary>Version comparison</summary>

```typ
#set page(width: auto)

#let test(v) = {
  import "@preview/fletcher:" + v: *

  [== #v]

  let n = node.with(shape: "rect", stroke: 0.5pt)

  diagram(
    debug: 1,
    axes: (ttb, ltr),
    spacing: (2em, 3em),
    {
      n((0, 0))[Left]
      n((0, 1))[Right]
    }
  )

  diagram(
    debug: 1,
    axes: (ttb, ltr),
    spacing: (2em, 3em),
    {
      n((0, 0))[Top]
      n((1, 0))[Bottom]
    }
  )

  diagram(
    debug: 1,
    axes: (ttb, ltr),
    spacing: (2em, 3em),
    {
      n((0, 0))[Top]
      n((0.5, 0.5))[aside]
      n((1, 0))[Bottom]
    }
  )

  diagram(
    debug: 1,
    axes: (ttb, ltr),
    spacing: (2em, 3em),
    {
      n((0, 0))[Left]
      n((0.5, 0.5))[aside]
      n((0, 1))[Right]
    }
  )
}

#table(
  columns: (5cm,)*6,
  gutter: 1cm,
  inset: 1em,
  // test("0.4.0"),
  test("0.4.1"),
  test("0.4.2"),
  test("0.4.3"),
  test("0.4.4"),
  // test("0.4.5"),
  test("0.5.0"),
  // test("0.5.1"),
  // test("0.5.2"),
  test("0.5.3"),
)
```
</details>

![image](https://github.com/user-attachments/assets/50f14639-0dc3-48db-be60-c58e2716f86f)

To summarize: when reversing the axes order
- 0.4.1 --> 0.4.2: only having one x/y coordinate would collapse nodes into the same y/x coordinate, while retaining the necessary height
- 0.4.2 -->  ... --> 0.5.0: similar effects with some differences in the details
- 0.5.2 --> 0.5.3: the width and height of nodes are confused when computing row and column dimensions

I did not test rtl or btt axes, but they may have similar problems
